lib/engine/game/g_1824.rb

```diff
@@ -33,6 +31,144 @@ class G1824 < Base
         'can_buy_p5' => ['Can buy P5', 'P5 can be bought']
       ).freeze
 
+      OPTIONAL_RULES = [
+        { sym: :cisleithania,
+          short_name: 'Cisleithania',
+          desc: 'Use the smaller Cislethania map, with some reduction of components - 2-3 players' },
+        { sym: :goods_time,
+          short_name: 'Goods Time',
+          desc: 'Use the Goods Time Variant (3-6 players) - pre-set scenario' },
+      ].freeze
+
+      CERT_LIMIT_CISLEITHANIA = { 2 => 14, 3 => 16 }.freeze
+
+      BANK_CASH_CISLEITHANIA = { 2 => 4000, 3 => 9000 }.freeze
+
+      CASH_CISLEITHANIA = { 2 => 830, 3 => 680 }.freeze
+
+      MOUNTAIN_RAILWAY_NAMES = {
+        1 => 'Semmeringbahn',
+        2 => 'Kastbahn',
+        3 => 'Brennerbahn',
+        4 => 'Arlbergbahn',
+        5 => 'Karawankenbahn',
+        6 => 'Wocheinerbahn',
+      }.freeze
+
+      def init_optional_rules(optional_rules)
+        opt_rules = super
+        validate_optional_rules(opt_rules)
+        opt_rules
+      end
+
+      def bank_cash
+        return super unless option_cislethania
+
+        BANK_CASH_CISLEITHANIA[@players.size] - @players.sum(&:cash)
+      end
+
+      def init_bank
+        return super unless option_cislethania
+
+        Engine::Bank.new(BANK_CASH_CISLEITHANIA[@players.size], log: @log)
+      end
+
+      def init_starting_cash(players, bank)
+        return super unless option_cislethania
+
+        players.each do |player|
+          bank.spend(CASH_CISLEITHANIA[@players.size], player)
+        end
+      end
+
+      def init_corporations(stock_market)
+        return super unless option_cislethania
+
+        # Remove Coal Railway C4 (SPB), Regional Railway BH and SB
+        CORPORATIONS.reject! { |c| %w[SPB SB BH].include?(c['sym']) }
```

Comment:
you should not mutate the constants! fix all of these instances

Reply:
fixed

Had to create location_name method to be able to handle LOCATION_NAMES changes.